docs/ERC20.md

```diff
@@ -0,0 +1,294 @@
+
+## Table of Contents
+
+- [Interface](#interface)
+  * [ERC20 compatibility](#erc20-compatibility)
+- [Usage](#usage)
+- [Extensibility](#extensibility)
+- [API Specification](#api-specification)
+  * [`name`](#-name-)
+  * [`symbol`](#-symbol-)
+  * [`decimals`](#-decimals-)
+  * [`total_supply`](#-total-supply-)
+  * [`balance_of`](#-balance-of-)
+  * [`allowance`](#-allowance-)
+  * [`transfer`](#-transfer-)
+  * [`transfer_from`](#-transfer-from-)
+  * [`approve`](#-approve-)
+
+## Interface
+
+```jsx
+@contract_interface
+namespace IERC20:
+		func name() -> (res: felt):
+    end
+
+    func symbol() -> (res: felt):
+    end
+
+    func decimals() -> (res: felt):
+    end
+
+    func total_supply() -> (res: Uint256):
+    end
+
+    func balance_of(account: felt) -> (res: Uint256):
+    end
+
+    func allowance(owner: felt, spender: felt) -> (res: Uint256):
+    end
+
+    func transfer(recipient: felt, amount: Uint256) -> (success: felt):
+    end
+
+    func transfer_from(
+            sender: felt, 
+            recipient: felt, 
+            amount: Uint256
+        ) -> (success: felt):
+    end
+
+    func approve(spender: felt, amount: Uint256) -> (success: felt):
+    end
+end
+```
+
+### ERC20 compatibility
+
+Although StarkNet is not EVM compatible, this implementation aims to be as close as possible to the ERC20 standard. This is why:
+
+- it uses Cairo's `uint256` instead of `felt`
+- it returns `1` as success to imitate a `bool`
+- it makes use of Cairo's short strings to simulate `name` and `symbol`
+- it will emit events once they're implemented on StarkNet.
+
+Notice that `get_decimals` returns a 252-bit `felt`, meaning it can be much larger than the standard's 8-bit `uint8` and this could potentially affect bridging tokens.
+
+## Usage
+
+Use cases go from medium of exchange currency to voting rights, staking, and more.
+
+Considering that the constructor method looks like this:
+
+```python
+func constructor(
+    name: felt,     # Token name as Cairo short string
+    symbol: felt,   # Token symbol as Cairo short string
+    recipient: felt # Address where to send initial supply to
+):
+```
+
+To create a token your need to deploy it like this:
+
+```python
+erc20 = await starknet.deploy(
+    "contracts/token/ERC20.cairo",
+    constructor_calldata=[
+        str_to_felt("Token"),    # name
+        str_to_felt("TKN"),      # symbol
+        account.contract_address # initial supply recipient
+    ]
+)
+```
+
+> Note that we haven't specified the initial supply. This is because it's fixed at `1000`, as well as `decimals` which is fixed at `18`. If you wish to change any of these values, you should modify the constructor.
+> 
+
+As most StarkNet contracts, it expects to be called by another contract and it identifies it through `get_caller_address` (analogous to Solidity's `msg.this`). This is why we need an Account contract to interact with it. For example:
+
+```python
+signer = Signer(123456789987654321)
```

Comment:
I still think this makes for confusing documentation. If it was something like `Signer.fromPrivateKey(1234...)` it would be clearer. Otherwise `Signer(PRIVATE_KEY)` might make it clearer, there is no need to define the variable even IMO.